Reduce Tech Debt and bring up to HQRM standards - Fixes #28

PlagueHO opened the pull request:
The purpose of the PR is to reduce the significant technical debt of this resource. This is so that new features can be added to the resource safely (#25, #26, #27).

The logic of the module is mostly unchanged, but about 60% of the lines of code had one or more guidelines violations which needed to be fixed.

The PR focuses mainly on:
- Improving code quality
- Upgrading code to meet standards (include HQRM)
- Completing unit and integration tests
- Fixing bugs (the xCertReq resource could not work at all in most environments in the current state).

The details of the PR are:
* Moved unit tests to correct folder structure.
* Changed unit tests to use standard test templates.
* Updated all resources to meet HQRM standards and style guidelines.
* Added .gitignore file
* Added .gitattributes file to force line endings to CRLF to allow unit tests to work.
* xCertificateCommon:
    - Moved common code into new module MSFT_xCertificateCommon.
    - Added standard exception code.
    - Renamed common functions Validate-* to use acceptable verb Test-*.
* xCertificateImport:
    - Fixed bug with Test-TargetResource incorrectly detecting change required.
    - Reworked unit tests for improved code coverage to meet HQRM standards.
    - Created Integration tests for both importing and removing an imported certificate.
    - Added descriptions to MOF file.
    - Removed default parameter values for parameters that are required or keys.
    - Added verbose messages.
    - Split message and error strings into localization string files.
* xPfxImport:
    - Fixed bug with Test-TargetResource incorrectly detecting change required.
    - Reworked unit tests for improved code coverage to meet HQRM standards.
    - Created Integration tests for both importing and removing an imported certificate.
    - Added descriptions to MOF file.
    - Removed default parameter values for parameters that are required or keys.
    - Added verbose messages.
    - Split message and error strings into localization string files.
* xCertReq:
    - Cleaned up descriptions in MOF file.
    - Fixed bugs generating certificate when credentials are specified.
    - Allowed output of certificate request when credentials are specified.
    - Split message and error strings into localization string files.
    - Created unit tests and integration tests.
    - Improved logging output to enable easier debugging.
* xPDT:
    - Renamed to match standard module name format (MSFT_x).
    - Modified to meet 100 characters or less line length where possible.
    - Split message and error strings into localization string files.

The only element that does not have unit test coverage is the MSFT_xPDT support module. This should be added when possible.

I apologize for the size of this change but it was the only way I could feel confident that the module was mostly free of tech debt (except for the missing unit tests on the xPDT support module).

The features and fixes in #25, #26 and #27 will begin to be submitted once this PR has moved through.

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/xcertificate/29)
<!-- Reviewable:end -->

msftclas commented:
Hi __@PlagueHO__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by Microsoft and real humans are currently evaluating your PR.</p>

TTYL, MSBOT;

PlagueHO commented:
@kwirkykat - do you know if anyone is actively managing this resource module over there at MSFT? Looks like it hasn't been active for a while.

kwirkykat commented:
Reviewed 3 of 39 files at r3.
Review status: 3 of 36 files reviewed at latest revision, 18 unresolved discussions.

---

*[ReadMe.md, line 110 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPqQvF8kempky47i4T:-KPPqQvF8kempky47i4U:batbpo8) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/ReadMe.md#L110)):*
> ```Markdown
> ## Examples
> 
> ## xCertReq
> ```

If Examples is now a 2-hash header, this should become a 3-hash header (###)

---

*[ReadMe.md, line 117 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPr0comCDfhUgKecYl:-KPPr0comCDfhUgKecYm:b-z4km2p) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/ReadMe.md#L117)):*
> ```Markdown
> configuration xCertReq_RequestSSL
> {
>     param (
> ```

Parenthesis should be on next line

---

*[ReadMe.md, line 121 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPr4-Y_YM2EokPtq77:-KPPr4-Y_YM2EokPtq78:b-cp39kt) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/ReadMe.md#L121)):*
> ```Markdown
>         [ValidateNotNullorEmpty()]
>         [PsCredential] $Credential
>         )
> ```

This parenthesis has one extra indent

---

*[ReadMe.md, line 153 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPqZhBtukp0c8bjyvE:-KPPqZhBtukp0c8bjyvF:batbpo8) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/ReadMe.md#L153)):*
> ```Markdown
> ```
> 
> ## xPfxImport
> ```

If Examples is now a 2-hash header, this should become a 3-hash header (###)

---

*[ReadMe.md, line 155 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPq_jhd4YJQPZAly7D:-KPPq_jhd4YJQPZAly7E:b-qnvox6) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/ReadMe.md#L155)):*
> ```Markdown
> ## xPfxImport
> 
> ### Simple Usage
> ```

If Examples is now a 2-hash header, this should become a 4-hash header (####)

---

*[ReadMe.md, line 160 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPqwaMSV-gIBfLbdkg:-KPPqwaMSV-gIBfLbdkh:b-z4km2p) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/ReadMe.md#L160)):*
> ```Markdown
> Configuration Sample_xPfxImport_MinimalUsage
> {
>     param(
> ```

Parenthesis should be on next line

---

*[ReadMe.md, line 185 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPqciN0ytxPqays7cV:-KPPqciN0ytxPqays7cW:b-qnvox6) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/ReadMe.md#L185)):*
> ```Markdown
> ```
> 
> ### Used with xWebAdministration Resources
> ```

If Examples is now a 2-hash header, this should become a 4-hash header (####)

---

*[ReadMe.md, line 190 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPqmjy1hpVIy-zaaQY:-KPPqmjy1hpVIy-zaaQZ:b-z4km2p) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/ReadMe.md#L190)):*
> ```Markdown
> Configuration Sample_xPfxImport_IIS_WebSite
> {
>     param(
> ```

Parenthesis should be on next line

---

*[ReadMe.md, line 239 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPqgbQ_Cf0e7UeNC5R:-KPPqgbQ_Cf0e7UeNC5S:batbpo8) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/ReadMe.md#L239)):*
> ```Markdown
> ```
> 
> ## xCertificateImport
> ```

If Examples is now a 2-hash header, this should become a 3-hash header (###)

---

*[DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1, line 46 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPrWL-lt43LJpC_sY1:-KPPrWL-lt43LJpC_sY2:b-z4km2p) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1#L46)):*
> ```PowerShell
> {
>     [CmdletBinding()]
>     param(
> ```

Parenthesis should be on next line

---

*[DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1, line 50 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPTc0ryT2J3Hr8h6cw9:-KPTc0rzcsKZKoXwlVGT:btr2jzc) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1#L50)):*
> ```PowerShell
>             Mandatory,
>             ValueFromPipeline
>         )]
> ```

This Parameter Attribute should all be on one line.

---

*[DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1, line 61 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPrdo1-QfXZtYMROrF:-KPPrdo1-QfXZtYMROrG:b-w43lql) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1#L61)):*
> ```PowerShell
>     Process
>     {
>         foreach ($p in $Path)
> ```

Please rename $p to something more descriptive (e.g. $singlePath, $individualPath, $certificatePath, etc.)

---

*[DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1, line 110 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPrZkbAqPDdRza0-et:-KPPrZkbAqPDdRza0-eu:b-z4km2p) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1#L110)):*
> ```PowerShell
> {
>     [CmdletBinding()]
>     param(
> ```

Parenthesis should be on next line

---

*[DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1, line 127 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPTc74uoCEPjRHRcdFq:-KPTc74uoCEPjRHRcdFr:bjkiaqv) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1#L127)):*
> ```PowerShell
>     {
>         # Get a list of all Valid Hash types and lengths into an array
>         $validHashes = [System.AppDomain]::CurrentDomain.GetAssemblies().GetTypes() |
> ```

As part of our new [Best Practices document](https://github.com/PowerShell/DscResources/blob/master/BestPractices.md#use-variables-rather-than-extensive-piping), we are asking that there only be at most one pipe in one line.
This is completely valid PowerShell, but to make the code easier to read/understand, can you please split these operations using variable assignment so that there is at most one pipe in one logical line of code?

---

*[DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1, line 173 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPtIW9qfmJBuiFadp1:-KPPtIWAVemRi0AAm_X5:b-mcq4dr) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1#L173)):*
> ```PowerShell
> } # end function Test-Thumbprint
> 
> function ThrowInvalidOperationError
> ```

Please update this function name to follow the Verb-Noun format with an approved PowerShell verb (e.g. New-InvalidOperationError)

Also, please add a help comment above this function.

---

*[DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1, line 190 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPtTdKh9ifuRWCoN5H:-KPPtTdLoPjmy1OWB7BR:bponsp9) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1#L190)):*
> ```PowerShell
> 
>     $exception = New-Object -TypeName System.InvalidOperationException `
>         -ArgumentList $ErrorMessage;
> ```

Please remove the semicolons at the end of the lines of this function

---

*[DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1, line 197 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPsLTuBCKDEEEEzA_s:-KPPsLTuBCKDEEEEzA_t:bz80esx) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1#L197)):*
> ```PowerShell
> } # end function ThrowInvalidOperationError
> 
> function ThrowInvalidArgumentError
> ```

Please update this function name to follow the Verb-Noun format with an approved PowerShell verb (e.g. New-InvalidArgumentError)

Also, please add a help comment above this function.

---

*[DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1, line 214 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPtZk0PMqzXNFtp86c:-KPPtZk0PMqzXNFtp86d:bponsp9) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1#L214)):*
> ```PowerShell
> 
>     $exception = New-Object -TypeName System.ArgumentException `
>         -ArgumentList $ErrorMessage;
> ```

Please remove the semicolons at the end of the lines of this function

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcertificate/29)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
Reviewed 2 of 39 files at r3.
Review status: 5 of 36 files reviewed at latest revision, 22 unresolved discussions.

---

*[DSCResources/MSFT_xCertificateImport/MSFT_xCertificateImport.psm1, line 32 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPTfkaeSYUgturmsmzo:-KPTfkafvTa8IbgmBaPf:b-m3b0j2) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateImport/MSFT_xCertificateImport.psm1#L32)):*
> ```PowerShell
>     param(
>         [Parameter(Mandatory)]
>         [ValidateScript( { $_ | Test-Thumbprint } )]
> ```

Where is the Test-Thumbprint function?

---

*[DSCResources/MSFT_xCertificateImport/MSFT_xCertificateImport.psm1, line 37 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPTfnJKKc9PGS_vve9b:-KPTfnJKKc9PGS_vve9c:b-nv1jrl) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateImport/MSFT_xCertificateImport.psm1#L37)):*
> ```PowerShell
> 
>         [Parameter(Mandatory)]
>         [ValidateScript( { $_ | Test-CertificatePath } )]
> ```

Where is the Test-CertificatePath function?

---

*[DSCResources/MSFT_xCertificateImport/MSFT_xCertificateImport.psm1, line 68 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPTh6ykh3f1K_CGT16x:-KPTh6ykh3f1K_CGT16y:bj7ik3x) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateImport/MSFT_xCertificateImport.psm1#L68)):*
> ```PowerShell
>     if ((Test-Path $CertificateStore) -eq $false)
>     {
>         ThrowInvalidArgumentError `
> ```

Where is ThrowInvalidArgumentError defined for this file?

---

*[DSCResources/MSFT_xCertificateImport/en-us/MSFT_xCertificateImport.strings.psd1, line 10 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPTgEl3fAhgPVbTsJ_y:-KPTgEl3fAhgPVbTsJ_z:bb8k4sr) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateImport/en-us/MSFT_xCertificateImport.strings.psd1#L10)):*
> ```PowerShell
> CertificateStoreNotFoundError=Certificate Store '{0}' not found.
> 
> ImportingCertificateShould=Importing certificate '{0}' into '{1}'
> ```

There should be one space on each side of the '='.
Also, can you please indent the text inside the here-string once (@' ... '@)?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcertificate/29)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Thanks for the great review @kwirkykat - sorry about all the mistakes. I made all the changes to this module before your other feed back on my other HQRM submitted modules so hadn't learned from those mistakes yet. I've now completed a whole lot more changes on this resource based on the comments you and the others made on my other PR's. Hopefully there won't be too many more mistakes.

It probably would have been easier and quicker to rewrite this resource from scratch. :cry:

There was also a lot of unused code in the xPDT module (I've no idea where this module came from but it must have been harvested from somewhere else). I've removed the unused code as I didn't see the point in documenting a bunch of functions that weren't being used (like NetUse - can't figure out why this was even included ever???).

Also, as we are doing all this work making this resource meet all the HQRM guidelines, is it worth renaming all the components be HQRM naming (e.g. CertificateDsc)? Or should we wait?

BTW, awesome job on open sourcing PS and getting it on Linux!

I'm done - have a great weekend!

---

Review status: 5 of 36 files reviewed at latest revision, 22 unresolved discussions, some commit checks failed.

---

*[ReadMe.md, line 110 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPqQvF8kempky47i4T:-KPWrN2wvmvgXAPPExIr:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/ReadMe.md#L110)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> If Examples is now a 2-hash header, this should become a 3-hash header (###)
</details>
Done.

---

*[ReadMe.md, line 117 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPr0comCDfhUgKecYl:-KPWrNbnZivNK-DIbSfu:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/ReadMe.md#L117)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Parenthesis should be on next line
</details>
Done.

---

*[ReadMe.md, line 121 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPr4-Y_YM2EokPtq77:-KPWrrsrvRkOhx4ZpvjU:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/ReadMe.md#L121)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> This parenthesis has one extra indent
</details>
Done.

---

*[ReadMe.md, line 153 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPqZhBtukp0c8bjyvE:-KPWrsvhf-UXNq0kPQX4:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/ReadMe.md#L153)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> If Examples is now a 2-hash header, this should become a 3-hash header (###)
</details>
Done.

---

*[ReadMe.md, line 155 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPq_jhd4YJQPZAly7D:-KPWrtNiJMdqBe622XtY:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/ReadMe.md#L155)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> If Examples is now a 2-hash header, this should become a 4-hash header (####)
</details>
Done.

---

*[ReadMe.md, line 160 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPqwaMSV-gIBfLbdkg:-KPWrtsMXs8i_M6yGvUA:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/ReadMe.md#L160)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Parenthesis should be on next line
</details>
Done.

---

*[ReadMe.md, line 185 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPqciN0ytxPqays7cV:-KPWruBuWrHmiqK0gACf:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/ReadMe.md#L185)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> If Examples is now a 2-hash header, this should become a 4-hash header (####)
</details>
Done.

---

*[ReadMe.md, line 190 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPqmjy1hpVIy-zaaQY:-KPWrutZ3iLG0-IXlWoW:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/ReadMe.md#L190)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Parenthesis should be on next line
</details>
Done.

---

*[ReadMe.md, line 239 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPqgbQ_Cf0e7UeNC5R:-KPWrwHs8qd5gsjr4khb:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/ReadMe.md#L239)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> If Examples is now a 2-hash header, this should become a 3-hash header (###)
</details>
Done.

---

*[DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1, line 46 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPrWL-lt43LJpC_sY1:-KPWsV1SrCi-sIqrk1mi:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1#L46)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Parenthesis should be on next line
</details>
Done.

---

*[DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1, line 50 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPTc0ryT2J3Hr8h6cw9:-KPWsVP7muL818b4eiFA:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1#L50)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> This Parameter Attribute should all be on one line.
</details>
Done.

---

*[DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1, line 61 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPrdo1-QfXZtYMROrF:-KPWuMpj8eT3A8_iDAUb:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1#L61)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Please rename $p to something more descriptive (e.g. $singlePath, $individualPath, $certificatePath, etc.)
</details>
Done.

---

*[DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1, line 110 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPrZkbAqPDdRza0-et:-KPWuOYVYFPDTm5S4zaG:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1#L110)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Parenthesis should be on next line
</details>
Done.

---

*[DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1, line 127 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPTc74uoCEPjRHRcdFq:-KPWuV-DuIPNhF7X7Cg7:b-4c8fa3) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1#L127)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> As part of our new [Best Practices document](https://github.com/PowerShell/DscResources/blob/master/BestPractices.md#use-variables-rather-than-extensive-piping), we are asking that there only be at most one pipe in one line.
> This is completely valid PowerShell, but to make the code easier to read/understand, can you please split these operations using variable assignment so that there is at most one pipe in one logical line of code?
</details>
Sounds like a pretty good practice to me. I'll remember it for next time :smile:  I've reworked this code so that it should be a lot easier to understand (hopefully).

---

*[DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1, line 173 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPtIW9qfmJBuiFadp1:-KPWzRG2rC7bthCtJwSj:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1#L173)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Please update this function name to follow the Verb-Noun format with an approved PowerShell verb (e.g. New-InvalidOperationError)
> 
> Also, please add a help comment above this function.
</details>
Done.

---

*[DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1, line 190 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPtTdKh9ifuRWCoN5H:-KPWxIMmJqCkhWh1hVt7:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1#L190)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Please remove the semicolons at the end of the lines of this function
</details>
Done.

---

*[DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1, line 197 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPsLTuBCKDEEEEzA_s:-KPWzRv-rWt8TOBvejO_:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1#L197)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Please update this function name to follow the Verb-Noun format with an approved PowerShell verb (e.g. New-InvalidArgumentError)
> 
> Also, please add a help comment above this function.
</details>
Done.

---

*[DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1, line 214 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPPtZk0PMqzXNFtp86c:-KPWxKO9OC7hd0j6r_D2:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateCommon/MSFT_xCertificateCommon.psm1#L214)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Please remove the semicolons at the end of the lines of this function
</details>
Done.

---

*[DSCResources/MSFT_xCertificateImport/MSFT_xCertificateImport.psm1, line 32 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPTfkaeSYUgturmsmzo:-KPWxSUGTe8v-pRT3kKK:bl2wq84) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateImport/MSFT_xCertificateImport.psm1#L32)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Where is the Test-Thumbprint function?
</details>
It's in the MSFT_xCertificateCommon module. All other modules import this module.

---

*[DSCResources/MSFT_xCertificateImport/MSFT_xCertificateImport.psm1, line 37 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPTfnJKKc9PGS_vve9b:-KPWxY5-RQtJGn0ucJL4:bl2wq84) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateImport/MSFT_xCertificateImport.psm1#L37)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Where is the Test-CertificatePath function?
</details>
It's in the MSFT_xCertificateCommon module. All other modules import this module.

---

*[DSCResources/MSFT_xCertificateImport/MSFT_xCertificateImport.psm1, line 68 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPTh6ykh3f1K_CGT16x:-KPWxhLhY8rXLArLacT6:bl2wq84) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateImport/MSFT_xCertificateImport.psm1#L68)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Where is ThrowInvalidArgumentError defined for this file?
</details>
It's in the MSFT_xCertificateCommon module. All other modules import this module.

---

*[DSCResources/MSFT_xCertificateImport/en-us/MSFT_xCertificateImport.strings.psd1, line 10 \[r3\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KPTgEl3fAhgPVbTsJ_y:-KPWy0nnMSwnDMJxq9kY:bd8jy1a) ([raw file](https://github.com/powershell/xcertificate/blob/6041f67444d880f812527564a389e73f70c24510/DSCResources/MSFT_xCertificateImport/en-us/MSFT_xCertificateImport.strings.psd1#L10)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> There should be one space on each side of the '='.
> Also, can you please indent the text inside the here-string once (@' ... '@)?
</details>
Fixed (in all other files too). Sorry about that.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcertificate/29#-:-KPWzUBJgn_uqFJHIdtP:bgmorot)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Hi @KwirkyKat - Is there anything you're waiting on from me for this one? My team would like to add some new resources and features to this repo (as per issues raised) but we're waiting for this PR to go through. No pressure :smile: 

---

Review status: 5 of 36 files reviewed at latest revision, 22 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcertificate/29#-:-KRFvCe-iZvk4-vECS55:b-bagfu1)*
<!-- Sent from Reviewable.io -->

kwirkykat commented:
@PlagueHO The MSFT_xCertificateCommon.psm1 module is not a resource, so it shouldn't follow the resource naming format. Can you rename this module to just CertificateCommon.psm1 or CommonResourceHelper.psm1 or something along those lines without the MSFT_x at the beginning?

Same thing goes for the xPDT module (even though that was mis-named before this PR). Since you're "renaming" it anyway, can you please change the name to just PDT.psm1 or somthing similar without the MSFT_x at the beginning since it is not a resource module? By the way, what does PDT stand for here?

There's also a lot of function help comments that aren't indented properly, but I won't hold such a large PR for just that.

That's it. Just the module renames and one place where you probably need to nullify the output and then this is good to go.

I don't see any obvious breaking changes here. Is there anything in this PR you would consider to be a breaking change?

---

Reviewed 8 of 39 files at r3, 14 of 25 files at r4, 1 of 4 files at r5, 1 of 2 files at r6, 1 of 1 files at r7.
Review status: 28 of 36 files reviewed at latest revision, 6 unresolved discussions.

---

*[DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1, line 28 \[r7\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KRGEOx7IbR85saaO9DI:-KRGEOx7IbR85saaO9DJ:b-ix82e9) ([raw file](https://github.com/powershell/xcertificate/blob/1f39ab0def1bd75d8542dec674fbdedf2c352b17/DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1#L28)):*
> ```PowerShell
> <#
>     .SYNOPSIS
>     Returns the current state of the certificate that may need to be requested.
> ```

Need an extra indent for everything under the help attributes

---

*[DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1, line 110 \[r7\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KRGENl3h9KyAMnQ5rrb:-KRGENl3h9KyAMnQ5rrc:b-ix82e9) ([raw file](https://github.com/powershell/xcertificate/blob/1f39ab0def1bd75d8542dec674fbdedf2c352b17/DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1#L110)):*
> ```PowerShell
> <#
>     .SYNOPSIS
>     Requests a new certificate based on the parameters provided.
> ```

Need an extra indent for everything under the help attributes

---

*[DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1, line 280 \[r7\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KRGE3ryL9gMM1f68iq9:-KRGE3ryL9gMM1f68iqA:b-p2ksh8) ([raw file](https://github.com/powershell/xcertificate/blob/1f39ab0def1bd75d8542dec674fbdedf2c352b17/DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1#L280)):*
> ```PowerShell
>             # a timing issue in MSFT_xPDT (the process has often completed before the
>             # process can be read in).
>             Start-Win32Process `
> ```

Can you add a $null = at the beginning of this line in case it does happen to have output?

---

*[DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1, line 355 \[r7\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KRGEHBJdBtS7Ulqmxfn:-KRGEHBKCB5CjINZqWHU:b-ix82e9) ([raw file](https://github.com/powershell/xcertificate/blob/1f39ab0def1bd75d8542dec674fbdedf2c352b17/DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1#L355)):*
> ```PowerShell
> <#
>     .SYNOPSIS
>     Tests if a new certificate should be requested.
> ```

Need an extra indent for everything under the help attributes

---

*[DSCResources/MSFT_xPDT/MSFT_xPDT.psm1, line 26 \[r7\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KRGFF2i2wwudR_UETC-:-KRGFF2i2wwudR_UETC0:b-ix82e9) ([raw file](https://github.com/powershell/xcertificate/blob/1f39ab0def1bd75d8542dec674fbdedf2c352b17/DSCResources/MSFT_xPDT/MSFT_xPDT.psm1#L26)):*
> ```PowerShell
> <#
>     .SYNOPSIS
>     Extracts an array of arguments that were found in the Arguments list passed in.
> ```

Need an extra indent for everything under the help attributes

---

*[Examples/Sample_xPfxImport_IIS_WebSite.ps1, line 6 \[r7\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KRGH_PD1XXiJS8n6aWo:-KRGH_PD1XXiJS8n6aWp:b939san) ([raw file](https://github.com/powershell/xcertificate/blob/1f39ab0def1bd75d8542dec674fbdedf2c352b17/Examples/Sample_xPfxImport_IIS_WebSite.ps1#L6)):*
> [PSCredential] $PfxPassword
[PSCredential] should be on its own line

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcertificate/29#-:-KRG9rnoUrzduSCcJEfX:b-x3kxn9)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
Awesome! Thank you @KwirkyKat.

I've renamed the MSFT_xPDT and xCertificateCommon to PDT and CertificateCommon respectively and moved them up into the DSCResources folder (no need for their own folders).

As for the name PDT: I have no idea either what it means to be honest - I know it back to front now but couldn't figure out what it meant. I'm assuming the P means either Process or PInvoke. I think @mgreenegit wrote it so he could probably tell us? :smile:

I've fixed the function help comment indent and spacing too because I want to make sure everything is as close to the guidelines as possible, but I must have missed those. Thanks for pointing them out.

I also quickly changed the Parameter definition layout in the Readme.md to show make the Key/Required/Write more visible (adopting the same layout I tend to use now).

One final thing I did was remove the DSCResourceDesigner_CreateScript.ps1 out of the examples folder as I don't think it is helpful and doesn't actually work and is also no longer matches the current module structure.

---

Review status: 28 of 36 files reviewed at latest revision, 6 unresolved discussions.

---

*[DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1, line 28 \[r7\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KRGEOx7IbR85saaO9DI:-KRGWsiNm-76UNO93pJl:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/1f39ab0def1bd75d8542dec674fbdedf2c352b17/DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1#L28)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Need an extra indent for everything under the help attributes
</details>
Done.

---

*[DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1, line 110 \[r7\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KRGENl3h9KyAMnQ5rrb:-KRGWt9S5a36iQEGPntK:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/1f39ab0def1bd75d8542dec674fbdedf2c352b17/DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1#L110)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Need an extra indent for everything under the help attributes
</details>
Done.

---

*[DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1, line 280 \[r7\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KRGE3ryL9gMM1f68iq9:-KRGX04C-LeCv0rWZCan:bkin9fa) ([raw file](https://github.com/powershell/xcertificate/blob/1f39ab0def1bd75d8542dec674fbdedf2c352b17/DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1#L280)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Can you add a $null = at the beginning of this line in case it does happen to have output?
</details>
Good thinking. Done. Also did the same for Wait-Win32ProcessEnd a few lines further on.

---

*[DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1, line 355 \[r7\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KRGEHBJdBtS7Ulqmxfn:-KRGX1QUnGCal_wwbgKm:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/1f39ab0def1bd75d8542dec674fbdedf2c352b17/DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1#L355)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Need an extra indent for everything under the help attributes
</details>
Done.

---

*[DSCResources/MSFT_xPDT/MSFT_xPDT.psm1, line 26 \[r7\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KRGFF2i2wwudR_UETC-:-KRGXBwYvAw7MqFuGT1o:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/1f39ab0def1bd75d8542dec674fbdedf2c352b17/DSCResources/MSFT_xPDT/MSFT_xPDT.psm1#L26)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> Need an extra indent for everything under the help attributes
</details>
Done.

---

*[Examples/Sample_xPfxImport_IIS_WebSite.ps1, line 6 \[r7\]](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KRGH_PD1XXiJS8n6aWo:-KRGZT-wLdWlsc0l883L:bsnackk) ([raw file](https://github.com/powershell/xcertificate/blob/1f39ab0def1bd75d8542dec674fbdedf2c352b17/Examples/Sample_xPfxImport_IIS_WebSite.ps1#L6)):*
<details><summary><i>Previously, kwirkykat (Katie Keim) wrote…</i></summary>
> > [PSCredential] $PfxPassword
> [PSCredential] should be on its own line
</details>
Done. And in the other Example and in the Readme.md

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcertificate/29#-:-KRGVMBHOBf8v4Wi51Qi:b-vpumpi)*
<!-- Sent from Reviewable.io -->

PlagueHO commented:
I've moved PDT and CertificateCommon modules back into their own folders (within DSCResources) because it then enables the standard unit test templates to test them. 

---

Review status: 28 of 36 files reviewed at latest revision, 6 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcertificate/29#-:-KRGa8QSl0uvDNGYUsa5:bovsnx6)*
<!-- Sent from Reviewable.io -->

narrieta commented:
Reviewed 2 of 44 files at r3, 17 of 17 files at r8.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1, line 282 at r8](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KShrYWtJtVwFSUzeiwF:-KShrYWtJtVwFSUzeiwG:b6sjovr) ([raw file](https://github.com/powershell/xcertificate/blob/1a318ced0691b9f40a705efc20b17f21d2f3297a/DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1#L282)):*
> $ENV:SystemRoot\System32\WindowsPowerShell\v1.0

Use $PSHOME instead

---

*[Examples/Sample_xPfxImport_MinimalUsage.ps1, line 26 at r8](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KShv8myu3STPBPBKexF:-KShv8mzKnTqjVopBOq4:bw6tj8p) ([raw file](https://github.com/powershell/xcertificate/blob/1a318ced0691b9f40a705efc20b17f21d2f3297a/Examples/Sample_xPfxImport_MinimalUsage.ps1#L26)):*
> ```PowerShell
> Start-DscConfiguration -Wait -Force -Verbose -Path 'c:\Sample_xPfxImport_MinimalUsage'
> 
> # Validate results
> ```

Add a comment of what the output of this command should be

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcertificate/29)*
<!-- Sent from Reviewable.io -->

narrieta commented:
Added 2 minor comments. Other than that it looks OK.

PlagueHO commented:
Thanks @narrieta ! I really appreciate you reviewing this (and the other reviewers in the DSC team). I've covered off the changes.

---

Review status: 34 of 36 files reviewed at latest revision, 8 unresolved discussions.

---

*[DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1, line 282 at r8](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KShrYWtJtVwFSUzeiwF:-KSjOgHnIBPegz_9B-7y:b-896fix) ([raw file](https://github.com/powershell/xcertificate/blob/1a318ced0691b9f40a705efc20b17f21d2f3297a/DSCResources/MSFT_xCertReq/MSFT_xCertReq.psm1#L282)):*
<details><summary><i>Previously, narrieta (Norberto Arrieta) wrote…</i></summary>
> > $ENV:SystemRoot\System32\WindowsPowerShell\v1.0
> 
> Use $PSHOME instead
</details>
Done.

---

*[Examples/Sample_xPfxImport_MinimalUsage.ps1, line 26 at r8](https://reviewable.io:443/reviews/powershell/xcertificate/29#-KShv8myu3STPBPBKexF:-KSjPednSIQqL9OKeXky:bjnv0wo) ([raw file](https://github.com/powershell/xcertificate/blob/1a318ced0691b9f40a705efc20b17f21d2f3297a/Examples/Sample_xPfxImport_MinimalUsage.ps1#L26)):*
<details><summary><i>Previously, narrieta (Norberto Arrieta) wrote…</i></summary>
> Add a comment of what the output of this command should be
</details>
I added the comment but I also updated the example so that it was actually easier to use (by parameterizing the variables in the config). I also improved the test so that it would only return the thumbprint that should have been imported (makes it much easier if you have lots of certs on your test machine).

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcertificate/29#-:-KSjRGj-mUqvjuOVvj8r:b3g9e0a)*
<!-- Sent from Reviewable.io -->

narrieta commented:
Reviewed 11 of 44 files at r3, 12 of 29 files at r4, 1 of 2 files at r6, 1 of 1 files at r7, 2 of 2 files at r9.
Review status: all files reviewed at latest revision, 5 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/xcertificate/29)*
<!-- Sent from Reviewable.io -->

narrieta commented:
@PlagueHO Thank you for your contribution!
